diff --git a/Utilities/bootstrap b/Utilities/bootstrap
index 5644301f..d9f43e88 100755
--- a/Utilities/bootstrap
+++ b/Utilities/bootstrap
@@ -215,7 +215,7 @@ def parse_build_args(args):
         if "macosx-arm64" in args.cross_compile_hosts:
             # Use XCBuild target directory when building for multiple arches.
             args.target_dir = os.path.join(args.build_dir, "apple/Products")
-        elif re.match('android-', args.cross_compile_hosts):
+        else:
             args.target_dir = os.path.join(
                                   args.build_dir,
                                   get_build_target(args,cross_compile=True))
@@ -792,10 +792,8 @@ def get_swiftpm_flags(args):
     cross_compile_hosts = args.cross_compile_hosts
     if build_target == 'x86_64-apple-macosx' and "macosx-arm64" in cross_compile_hosts:
         build_flags += ["--arch", "x86_64", "--arch", "arm64"]
-    elif cross_compile_hosts and re.match('android-', cross_compile_hosts):
-        build_flags.extend(["--destination", args.cross_compile_config])
     elif cross_compile_hosts:
-        error("cannot cross-compile for %s" % cross_compile_hosts)
+        build_flags.extend(["--destination", args.cross_compile_config])
 
     # Ensure we are not sharing the module cache with concurrent builds in CI
     local_module_cache_path=os.path.join(args.build_dir, "module-cache")
