diff --git a/Utilities/bootstrap b/Utilities/bootstrap
index 92c368a86..ac68540e6 100755
--- a/Utilities/bootstrap
+++ b/Utilities/bootstrap
@@ -215,7 +215,7 @@ def parse_build_args(args):
         if re.match("macosx-", args.cross_compile_hosts):
             # Use XCBuild target directory when building for multiple arches.
             args.target_dir = os.path.join(args.build_dir, "apple/Products")
-        elif re.match('android-', args.cross_compile_hosts):
+        else:
             args.target_dir = os.path.join(
                                   args.build_dir,
                                   get_build_target(args,cross_compile=True))
@@ -833,7 +833,7 @@ def get_swiftpm_flags(args):
         elif re.match('android-', cross_compile_hosts):
             build_flags.extend(["--destination", args.cross_compile_config])
         else:
-            error("cannot cross-compile for %s" % cross_compile_hosts)
+            build_flags.extend(["--destination", args.cross_compile_config])
 
     # Ensure we are not sharing the module cache with concurrent builds in CI
     local_module_cache_path=os.path.join(args.build_dir, "module-cache")
