diff --git a/Utilities/build-script-helper.py b/Utilities/build-script-helper.py
index 7c21a7b..493c78b 100755
--- a/Utilities/build-script-helper.py
+++ b/Utilities/build-script-helper.py
@@ -90,11 +90,9 @@ def get_swiftpm_options(swift_exec, args):
   if args.cross_compile_host:
     if build_target == 'x86_64-apple-macosx' and args.cross_compile_host == "macosx-arm64":
       swiftpm_args += ["--arch", "x86_64", "--arch", "arm64"]
-    elif re.match('android-', args.cross_compile_host):
+    else:
       print('Cross-compiling for %s' % args.cross_compile_host)
       swiftpm_args += ['--destination', args.cross_compile_config]
-    else:
-      error("cannot cross-compile for %s" % args.cross_compile_host)
 
   return swiftpm_args
 
