diff --git a/Utilities/build-script-helper.py b/Utilities/build-script-helper.py
index bd31bec..c73bba2 100755
--- a/Utilities/build-script-helper.py
+++ b/Utilities/build-script-helper.py
@@ -139,11 +139,9 @@ def get_swiftpm_options(swift_exec: str, args: argparse.Namespace) -> List[str]:
     if args.cross_compile_host:
         if re.search('-apple-macosx', build_target) and re.match('macosx-', args.cross_compile_host):
             swiftpm_args += ["--arch", "x86_64", "--arch", "arm64"]
-        elif re.match('android-', args.cross_compile_host):
+        else:
             print('Cross-compiling for %s' % args.cross_compile_host)
             swiftpm_args += ['--destination', args.cross_compile_config]
-        else:
-            fatal_error("cannot cross-compile for %s" % args.cross_compile_host)
 
     return swiftpm_args
 
